### PR TITLE
Adds eslint, eslint-config-vuepress for use in vuepress-theme-odyssey

### PR DESCRIPTION
**Pull request by arnoldsandoval-okta**
Adds eslint to monorepo, [eslint-config-vuepress](https://www.npmjs.com/package/eslint-config-vuepress) for use in vuepress-theme-odyssey (DRAFT PR is available [here](https://github.com/okta/odyssey/pull/667))

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -49,6 +49,15 @@
     jsesc "^2.5.1"
     source-map "^0.5.0"
 
+"@babel/generator@^7.11.5":
+  version "7.11.6"
+  resolved "https://artifacts.aue1d.saasure.com:443/artifactory/api/npm/npm-okta-master/@babel/generator/-/generator-7.11.6.tgz#b868900f81b163b4d464ea24545c61cbac4dc620"
```
Comment (jmelberg-okta) [inline yarn.lock:54]: This should be our public registry - not internal.

File: packages/vuepress-theme-odyssey/.eslintignore [hunk 2]
```diff
@@ -0,0 +1 @@
+!.vuepress
```
Comment (jmelberg-okta) [inline packages/vuepress-theme-odyssey/.eslintignore:1]: Is the follow-up PR going to remove this and apply the rule fixes?

File: packages/vuepress-theme-odyssey/package.json [hunk 3]
```diff
@@ -5,10 +5,15 @@
   "author": "Okta, Inc.",
   "license": "Apache-2.0",
   "main": "index.js",
+  "scripts": {
+    "lint": "eslint . --ext .js,.vue",
```
Comment (jmelberg-okta) [inline packages/vuepress-theme-odyssey/package.json:9]: Can the `--ext` values be moved to our config file?

**Review by jmelberg-okta — APPROVED** [review]
Couple of questions.

Please make sure to address the `yarn.lock` changes before merging.

File: yarn.lock (hunk 1 above)
Comment (arnoldsandoval-okta) [reply yarn.lock:54]: Will fix.

File: packages/vuepress-theme-odyssey/.eslintignore (hunk 2 above)
Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/.eslintignore:1]: This actually shouldn't be here at all. Will remove.

File: packages/vuepress-theme-odyssey/package.json (hunk 3 above)
Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/package.json:9]: This doesnt appear to be possible, I did a little digging and found this, but its fairly old:

 https://github.com/eslint/eslint/issues/11223

Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/package.json:9]: Spoke to @jmelberg-okta offline, he referenced this RFC - but work does not been merged for this specific request (as far as I was able to tell) https://github.com/eslint/rfcs/pull/9

Looking at how okta-vue does it, they have taken the same approach as above. https://github.com/okta/okta-vue/blob/master/package.json#L14

In the short term, will not fix

**Pull request merged**